src/HttpOverStream.Server.Owin/CustomListenerHost.cs

```diff
@@ -41,31 +41,40 @@ private async void OnAccept(Stream stream)
             {
                 using (stream)
                 {
+                    var sendHeadersCallback = new List<(Action<object>, object)>();
                     var owinContext = new OwinContext();
+                    // this is an owin extension
+                    owinContext.Set<Action<Action<object>, object>>("server.OnSendingHeaders", (callback, state) =>
+                   {
+                       sendHeadersCallback.Add((callback, state));

```

Comment:
This code looks like it adds a callback every time the headers are sent, this is not the case right?

Reply:
Nope, this code setup a callback recorder (an owin middleware can register itself to be notified when the headers are about to be sent - to manipulate headers).